.circleci/config.yml

```
@@ -317,34 +324,44 @@ jobs:
-          at: .
-      - run: pip install coverage codecov diff_cover
-      - run: ls -hal *.coverage
-      # Combine all job coverage reports into one
-      - run: coverage combine *.coverage
-      # Upload coverage report to Codecov
-      # DEV: Do not use the bash uploader, it cannot be trusted
-      - run: codecov
-      # Generate and save xml report
-      # DEV: "--ignore-errors" to skip over files that are missing
-      - run: coverage xml --ignore-errors
-      - store_artifacts:
-          path: coverage.xml
-      # Generate and save JSON report
-      # DEV: "--ignore-errors" to skip over files that are missing
-      - run: coverage json --ignore-errors
-      - store_artifacts:
-          path: coverage.json
-      # Print ddtrace/ report to stdout
-      # DEV: "--ignore-errors" to skip over files that are missing
-      - run: coverage report --ignore-errors --omit=tests/
-      # Print tests/ report to stdout
-      # DEV: "--ignore-errors" to skip over files that are missing
-      - run: coverage report --ignore-errors --omit=ddtrace/
-      # Print diff-cover report to stdout (compares against origin/1.x)
-      - run: diff-cover --compare-branch $(git rev-parse --abbrev-ref origin/HEAD) coverage.xml
+      - when:
+          condition:
+            << pipeline.parameters.coverage >>
+          steps:
+            - checkout
+            - attach_workspace:
+                at: .
+            - run: pip install coverage codecov diff_cover
+            - run: ls -hal *.coverage
+            # Combine all job coverage reports into one
+            - run: coverage combine *.coverage
+            # Upload coverage report to Codecov
+            # DEV: Do not use the bash uploader, it cannot be trusted
+            - run: codecov
+            # Generate and save xml report
+            # DEV: "--ignore-errors" to skip over files that are missing
+            - run: coverage xml --ignore-errors
+            - store_artifacts:
+                path: coverage.xml
+            # Generate and save JSON report
+            # DEV: "--ignore-errors" to skip over files that are missing
+            - run: coverage json --ignore-errors
+            - store_artifacts:
+                path: coverage.json
+            # Print ddtrace/ report to stdout
+            # DEV: "--ignore-errors" to skip over files that are missing
+            - run: coverage report --ignore-errors --omit=tests/
+            # Print tests/ report to stdout
+            # DEV: "--ignore-errors" to skip over files that are missing
+            - run: coverage report --ignore-errors --omit=ddtrace/
+            # Print diff-cover report to stdout (compares against origin/1.x)
+            - run: diff-cover --compare-branch $(git rev-parse --abbrev-ref origin/HEAD) coverage.xml
+      - unless:
+          condition:
+            << pipeline.parameters.coverage >>
+          steps:
+            - run: echo "No coverage on non-1.x branches"
```

```suggestion
            - run: echo "No coverage data collected. Set run workflow manually with pipeline parameter 'coverage=true'."
```

or something... wdyt?